tests/test_ptycho.py

```diff
@@ -49,6 +49,105 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 
+import lzma
+import numpy as np
+import os
+import pickle
+import tike.ptycho
+import unittest
+
 __author__ = "Daniel Ching"
 __copyright__ = "Copyright (c) 2016, UChicago Argonne, LLC."
 __docformat__ = 'restructuredtext en'
+
+
+class TestPtychoRecon(unittest.TestCase):
+    """Test various ptychography reconstruction methods for consistency."""
+
+    def create_dataset(self, dataset_file):
+        """Create a dataset for testing this module."""
+        import matplotlib.pyplot as plt
+        amplitude = plt.imread("./tests/data/Cryptomeria_japonica-0128.tif")
```

Comment:
I'd avoid having dependency to matplotlib just for reading tifs. Use numpy instead.

Reply:
`numpy` can't load tiffs, so I was trying to avoid data redundancy by loading the tiffs from the demo notebooks. If we store all data as `pickle` or `hdf5`, then `matplotlib` will only be a dependency for `tike.view`. `hdf5` is probably the better choice because it is compatible with `mpi` parallelism. The drawback is that all data will be uncompressed.

Reply:
But actually, all of our unit tests will be run on single cores? So `mpi` isn't an option anyways? What do you think?

Reply:
Actually, I realized that `matplotlib` isn't required to run these tests. That's why the import statement is in this function instead of at the top of the module. It's only required to regenerate the dataset if `setUp` detects that `ptycho_setup.pickle.lzma` has been deleted.